File: 240425/treemap 기본/treemap-basic.py
from sortedcontainers import SortedDict

n = int(input())
arr = SortedDict()
for _ in range(n):
    x = list(input().split())

    if x[0] == 'add':
        arr[int(x[1])] = int(x[2])
    elif x[0] == 'remove':
        arr.pop(int(x[1]))
    elif x[0] == 'find':
        if int(x[1]) in arr:
            print(arr[int(x[1])])
        else:
            print(None)
    elif x[0] == 'print_list':
        if arr:
            for v in arr.values():
                print(int(v), end = " ")
            print()
        else:
            print(None)

File: 240629/treemap 기본/treemap-basic.js
const fs = require("fs")
const input = fs.readFileSync(0).toString().trim().split("\n")
const SortedMap = require("collections/sorted-map");

const n = Number(input[0])

const sm = new SortedMap()

for (let i = 1; i <= n; i++) {
    let line = input[i].split(" ")
    if (line[0] === "add") {
        sm.set(parseInt(line[1]), parseInt(line[2]))
    } else if (line[0] === "remove") {
        sm.delete(parseInt(line[1]))
    } else if (line[0] === "find") {
        if (sm.has(parseInt(line[1]))) {
            console.log(sm.get(parseInt(line[1])))
        } else {
            console.log("None")
        }
    } else {
        if (sm.length === 0) {
            console.log("None")
        } else {
            total = ""
            for (let [k, v] of sm.entries()) {
                total += v + " "
            }
            console.log(total)
        }
    }
}